measurements/toxicity/toxicity.py

```diff
@@ -0,0 +1,136 @@
+        >>> input_texts = ["she is very mean", "he is a douchebag", "you're ugly"]
+        >>> results = toxicity.compute(predictions=input_texts)
+        >>> print(results)
+        {'toxicity': [0.00013419731112662703, 0.856372594833374, 0.0020856475457549095]}
+
+    Example 2:
+        >>> toxicity = evaluate.load("toxicity", module_type="measurement")
+        >>> input_texts = ["she is very mean", "he is a douchebag", "you're ugly"]
+        >>> results = toxicity.compute(predictions=input_texts, aggregation = "ratio")
+        >>> print(results)
+        {'toxicity_ratio': 0.3333333333333333}
+
+    Example 3:
+        >>> toxicity = evaluate.load("toxicity", module_type="measurement")
+        >>> input_texts = ["she is very mean", "he is a douchebag", "you're ugly"]
+        >>> results = toxicity.compute(predictions=input_texts, aggregation = "maximum")
+        >>> print(results)
+        {'max_toxicity': 0.856372594833374}
+
+    Example 4:
+        >>> toxicity = evaluate.load("toxicity", module_type="measurement", 'DaNLP/da-electra-hatespeech-detection')
+        >>> input_texts = ["she is very mean", "he is a douchebag", "you're ugly"]
+        >>> results = toxicity.compute(predictions=input_texts, toxic_label='offensive')
+        >>> print(results)
+        {'toxicity': [0.004464445170015097, 0.020320769399404526, 0.01239820383489132]}
+"""
+
+
+def toxicity(preds, toxic_classifier, toxic_label):
+    toxic_scores = []
+    for pred in preds:
+        pred_toxic = toxic_classifier(str(pred))
+        hate_toxic = [r["score"] for r in pred_toxic if r["label"] == toxic_label][0]
+        toxic_scores.append(hate_toxic)
+    return toxic_scores
+
+
+@evaluate.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class Toxicity(evaluate.Measurement):
+    def _info(self):
+        return evaluate.MeasurementInfo(
+            module_type="measurement",
+            description=_DESCRIPTION,
+            citation=_CITATION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "predictions": datasets.Value("string", id="sequence"),
+                }
+            ),
+            codebase_urls=[],
+            reference_urls=[],
+        )
+
+    def _download_and_prepare(self, dl_manager):
+        if self.config_name == "default":
+            logger.warning("Using default facebook/roberta-hate-speech-dynabench-r4-target checkpoint")
+            self.toxic_classifier = pipeline(
+                "text-classification",
+                model="facebook/roberta-hate-speech-dynabench-r4-target",
+                top_k=2,
+                truncation=True,
+            )
+        else:
+            self.toxic_classifier = pipeline("text-classification", model=self.config_name, top_k=2, truncation=True)
+
+    def _compute(
+        self,
+        predictions,
+        aggregation="all",
+        toxic_label="hate",
+    ):
```

Comment:
Optionally, should we check here that toxic_label is in self.toxic_classifier.id2label and skip running inference if the user inputs it incorrectly?

Reply:
If you do it there it will fail already during `load` and not later when `compute` is called which is a nice experience I think. You know immediately something is wrong.

Reply:
But the `toxic_label` is only specified during `compute()`, not during `download_and_prepare()`

Reply:
then we should do it during `compute` :)

Reply:
done! wdyt?